vault/backend.go

```diff
@@ -33,6 +37,30 @@ type listedAccount struct {
 	WatchOnly        bool   `json:"watch_only"`
 }
 
+type backend struct {
+	*framework.Backend
+}
+
+func (b *backend) importNode(ctx context.Context, req *logical.Request,
+	data *framework.FieldData) (*logical.Response, error) {
+
+	strNode := data.Get("node").(string)
+	strNet := data.Get("network").(string)
+
+	seed, err := seedFromSeedAndPassPhrases(
+		data.Get("seedphrase").(string),
+		data.Get("passphrase").(string),
+	)
+	if err != nil {
+		b.Logger().Error("Failed to get seed from seed and "+
+			"pass phrases", "error", err)
+		return nil, err
+	}
+	defer zero(seed)
```

Comment:
You're right. I changed all of the handling to use byte slices and arrays, which I am able to zero inside the function. I can't do anything about the strings passed as arguments, but I can zero everything under my control.

Reply:
Is it worth doing anything at all then?

Reply:
I think having fewer copies of the private key material in memory is worth it. The derived private keys are also zeroed at every opportunity, which is what btcwallet does as well. The code complexity isn't much worse with byte slices because the same Split utility exists as with strings, but it returns sub-slices rather than copies. I only ever need to zero the underlying slice and all the sub-slices get zeroed for free.

I've now moved the string-to-byte-slice conversion to where the string originates and made `seedFromSeedAndPassPhrases` accept a byte slice which it zeroes. Also, I just tried to use a byte slice (with a `TypeSlice` field type) instead of string as an argument to see if I could get it to zero my request in the unit tests, and that didn't work. This might be an opportunity to improve vault, so I added issue #20 to investigate further.

As an aside, the reason I use a storage backend in the tests here, even though it's slower, is that when I use an in-memory backend and zero a secret, it gets zeroed in the storage engine. So I think vault is pretty good about not making copies of byte slices internally when they're in the secrets engine, I think this is just an issue with API requests.

Reply:
Also found https://github.com/golang/go/issues/21865 - maybe `zero()` isn't sufficient either?

Reply:
I've removed all new use of `zero()` from this PR and specifically this commit. I'll do further investigation under #20 to try to improve this sustainably, maybe write some tests using `gcore` to automatically detect leftover key material.